### DISCUSSION TITLE
Computed properties on Form Object

### DISCUSSION THREAD

**Discussion by tekrai** [livewire/livewire#7161] [1. Help]
Hi,

**What I'm using :**

- php 8.2.10
- Livewire 3.0.5

**My problem :**
I'm trying to set a computed property directly on a Form object, without success.

Does someone know how to fix this ? 

```php

use Livewire\Form;

class DatePicker extends Form
{
    #[Rule('required|date_format:d/m/Y')]
    public string $from_date;

    #[Rule('required|date_format:d/m/Y|after_or_equal:from_date')]
    public string $to_date;

    #[Computed]
    public function period()
    {
        // some code to return the date range selected.
    }
}
```

```php

use Livewire\Component;

class SomePage extends Component
{

    public DatePicker $datePicker;

    public function mount(): void
    {
        $this->datePicker->setFilters();
        $this->datePicker->period; // [ErrorException] Undefined property: App\Livewire\Forms\DatePicker::$period
    }
}
```

It would be awesome if it worked, as I wouldn't have to repeat myself inside each Livewire component where the DatePicker would be used.

For now, I'm calling the method directly.

`$this->datePicker->period()`

Thank you for reading this and for your help. I'm sure it's something simple that I missed 😅

**Comment by satoved** [comment]
What's wrong with calling method directly in your case?

**Comment by tekrai** [comment]
Hi,

Thank you for your response.
You're right; I can call the method directly without any issues.
Since the result of this method remains consistent for each Livewire cycle, I wanted to cache it, similar to how one would with a typical computed property in a Livewire Object.
I've placed the logic within the form itself to reduce code boilerplate in my component.
